spec/src/antlr/JDQL.g4

```diff
@@ -0,0 +1,93 @@
+grammar JDQL;
+
+statement : select_statement | update_statement | delete_statement;
+
+select_statement : select_clause? from_clause? where_clause? orderby_clause?;
```

Comment:
@gavinking this statement has a bug because an empty query "" is valid.

We need to elaborate on this point more:

```antlr
mandatory_statement : select_clause from_clause where_clause orderby_clause
                    | select_clause from_clause where_clause
                    | select_clause from_clause orderby_clause
                    | select_clause where_clause orderby_clause
                    | from_clause where_clause orderby_clause
                    | where_clause
                    ;

select_statement : mandatory_statement
                 | select_clause
                 | from_clause
                 | where_clause
                 | orderby_clause
                 ;
```

Reply:
Well the idea here is that:

```java
@Query("")
List<Book> allBooks();
```

is a legal query with a clear interpretation.

Reply:
Yes of course. I'm not advocating it as the best way to write that code; just saying that it's consistent and "not wrong".

Reply:
[Recall also that the eventual goal of this work is to redefine other things like `@Find` _in terms of JDQL_.]